packages/flutter/test/material/bottom_navigation_bar_test.dart

```diff
@@ -100,19 +100,99 @@ void main() {
+    const TextStyle selectedTextStyle = TextStyle(fontWeight: FontWeight.w200, fontSize: 18.0);
+    const TextStyle unselectedTextStyle = TextStyle(fontWeight: FontWeight.w600, fontSize: 12.0);
+
+    await tester.pumpWidget(
+        MaterialApp(
+          home: Scaffold(
+            bottomNavigationBar: BottomNavigationBar(
+              type: BottomNavigationBarType.fixed,
+              selectedLabelStyle: selectedTextStyle,
+              unselectedLabelStyle: unselectedTextStyle,
+              items: const <BottomNavigationBarItem>[
+                BottomNavigationBarItem(
+                  icon: Icon(Icons.ac_unit),
+                  title: Text('AC'),
+                ),
+                BottomNavigationBarItem(
+                  icon: Icon(Icons.access_alarm),
+                  title: Text('Alarm'),
+                ),
+              ],
+            ),
+          ),
+        )
+    );
+
+    final TextStyle selectedFontStyle = tester.renderObject<RenderParagraph>(find.text('AC')).text.style;
+    final TextStyle unselectedFontStyle = tester.renderObject<RenderParagraph>(find.text('Alarm')).text.style;
+    expect(selectedFontStyle.fontSize, equals(selectedTextStyle.fontSize));
+    expect(selectedFontStyle.fontWeight, equals(selectedTextStyle.fontWeight));
+    expect(
+      tester.firstWidget<Transform>(find.ancestor(of: find.text('Alarm'), matching: find.byType(Transform))).transform,
+      equals(Matrix4.diagonal3(Vector3.all(unselectedTextStyle.fontSize / selectedTextStyle.fontSize))),
+    );
+    expect(unselectedFontStyle.fontWeight, equals(unselectedTextStyle.fontWeight));
+  });
+
+  testWidgets('font size on text styles overrides font size params', (WidgetTester tester) async {
+    const TextStyle selectedTextStyle = TextStyle(fontSize: 18.0);
+    const TextStyle unselectedTextStyle = TextStyle(fontSize: 12.0);
+    const double selectedFontSize = 17.0;
+    const double unselectedFontSize = 11.0;
+
+    await tester.pumpWidget(
+        MaterialApp(
+          home: Scaffold(
+            bottomNavigationBar: BottomNavigationBar(
+              type: BottomNavigationBarType.fixed,
+              selectedLabelStyle: selectedTextStyle,
+              unselectedLabelStyle: unselectedTextStyle,
+              selectedFontSize: selectedFontSize,
+              unselectedFontSize: unselectedFontSize,
+              items: const <BottomNavigationBarItem>[
+                BottomNavigationBarItem(
+                  icon: Icon(Icons.ac_unit),
+                  title: Text('AC'),
+                ),
+                BottomNavigationBarItem(
+                  icon: Icon(Icons.access_alarm),
+                  title: Text('Alarm'),
+                ),
+              ],
+            ),
+          ),
+        )
+    );
+
+    final TextStyle selectedFontStyle = tester.renderObject<RenderParagraph>(find.text('AC')).text.style;
+    expect(selectedFontStyle.fontSize, equals(selectedTextStyle.fontSize));
+    expect(
```

Comment:
Why does this test look at a transform matrix? Is there a less implementation-specific way to test the font size ratio?

Reply:
We have to do this because the fontsize for unselected text items is never actually updated. The animations are accomplished smoothly by applying a transform to the text widgets. 

The only way that I know of to verify the size of the unselected text items is to verify that the transform is correctly applied.

Reply:
I wish they were just lerping.